compiler-core/src/language_server/code_action.rs

```diff
@@ -75,6 +75,37 @@ impl CodeActionBuilder {
     }
 }
 
+/// A little wrapper around LineNumbers to make it easier to build text edits.
+///
+struct LineNumbersHelper<'a> {
+    line_numbers: &'a LineNumbers,
+}
+
+impl<'a> LineNumbersHelper<'a> {
+    fn new(line_numbers: &'a LineNumbers) -> Self {
+        LineNumbersHelper { line_numbers }
+    }
+
+    fn src_span_to_lsp_range(&self, location: SrcSpan) -> Range {
+        src_span_to_lsp_range(location, self.line_numbers)
+    }
+
+    fn edit_src_span(&self, location: SrcSpan, new_text: String) -> TextEdit {
```

Comment:
The names of the method in this class could be more consistent. Perhaps `insert`, `replace`, and `delete`?

```rust
self.edits.delete(some_span);
```
```rust
self.edits.replace(some_span, some_content);
```

The class could also hold the edits vector.

Could you make the methods used outside this class public please? To make the intended API more clear.